GitCommands/PathUtil.cs

```diff
@@ -124,14 +124,42 @@ public static string NormalizePath([NotNull] this string path)
 
             try
             {
-                return Path.GetFullPath(new Uri(path).LocalPath);
+                return Path.GetFullPath(Resolve(path));
             }
             catch (UriFormatException)
             {
                 return string.Empty;
             }
         }
 
+        [NotNull]
+        public static string Resolve([NotNull] string path, string relativePath = "")
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new ArgumentException(nameof(path));
+            }
+
+            // Fix for on purpose invalid machine name of WSL in Windows 10
+            bool replaceWslPath = path.StartsWith(@"\\wsl$\");
+            if (replaceWslPath)
+            {
+                // Temporarily replace machine name with a valid name
+                path = Regex.Replace(path, "^" + Regex.Escape(@"\\wsl$\"), @"\\wsl\");
```

Comment:
Since we know the path starts with the specified string, why do we need to use regex? Why can't we just use `string.Replace`?

Regex is *quite* expensive, it has been [improved in .NET5](https://devblogs.microsoft.com/dotnet/regex-performance-improvements-in-net-5/), but even then it is probably an overkill...